### PR TITLE
refactor: Extract Tag unused style

### PR DESCRIPTION
**Pull request by zombieJ**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      Extract Tag status & preset color style which will only generate by needed.     |
| 🇨🇳 Chinese |      剥离 Tag 状态与预设颜色样式，现在只有当使用的使用才会生成它们。       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 65a8e96</samp>

Refactored the tag component and its style logic to use sub-style components for preset and status colors, and fixed some issues with the button component. Renamed `CompactStyle` to `CompactCmp` and added a `key` prop in `components/button/button.tsx`. Extracted the preset and status styles into separate files and functions in `components/tag/style/`. Updated the `genSubStyleComponent` function in `components/theme/util/genComponentStyleHook.ts` to accept more parameters and set default options.

### <samp>🤖 Generated by Copilot at 65a8e96</samp>

*  Rename `CompactStyle` to `CompactCmp` and add `key` prop to avoid confusion and warning ([link](https://github.com/ant-design/ant-design/pull/44512/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140L28-R28), [link](https://github.com/ant-design/ant-design/pull/44512/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140L292-R292))
*  Reorder imports of `React`, `CloseOutlined` and `classNames` to match `components/button/button.tsx` ([link](https://github.com/ant-design/ant-design/pull/44512/files?diff=unified&w=0#diff-bcceb016a2b5f77c9c373d92bb6e81d64f26323154b13adfc6b74243e78b4913L3-R6), [link](https://github.com/ant-design/ant-design/pull/44512/files?diff=unified&w=0#diff-cb2262b77ae285e7da02504971f378bec871797a530abbf2a34f14fdb3d92f85L1-R8))
*  Import `PresetCmp` and `StatusCmp` from sub-style files to render preset and status styles of tag component ([link](https://github.com/ant-design/ant-design/pull/44512/files?diff=unified&w=0#diff-bcceb016a2b5f77c9c373d92bb6e81d64f26323154b13adfc6b74243e78b4913R16-R17))
*  Add `isPreset` and `isStatus` variables to store results of color check functions and improve readability ([link](https://github.com/ant-design/ant-design/pull/44512/files?diff=unified&w=0#diff-bcceb016a2b5f77c9c373d92bb6e81d64f26323154b13adfc6b74243e78b4913L72-R77))
*  Add `PresetCmp` and `StatusCmp` to tag component's children based on `color` prop ([link](https://github.com/ant-design/ant-design/pull/44512/files?diff=unified&w=0#diff-bcceb016a2b5f77c9c373d92bb6e81d64f26323154b13adfc6b74243e78b4913R148-R149))
*  Export `TagToken` interface to be used by sub-style components and files ([link](https://github.com/ant-design/ant-design/pull/44512/files?diff=unified&w=0#diff-cb2262b77ae285e7da02504971f378bec871797a530abbf2a34f14fdb3d92f85L21-R23))
*  Move `genTagStatusStyle` and `genPresetStyle` functions to sub-style files to modularize tag style logic ([link](https://github.com/ant-design/ant-design/pull/44512/files?diff=unified&w=0#diff-cb2262b77ae285e7da02504971f378bec871797a530abbf2a34f14fdb3d92f85L31-L68), [link](https://github.com/ant-design/ant-design/pull/44512/files?diff=unified&w=0#diff-79f09cde60f663f18ad398ac97f9dffdd51e3ff6c851cfc508e5cfdb51cc11d9R1-R33), [link](https://github.com/ant-design/ant-design/pull/44512/files?diff=unified&w=0#diff-b41be4b94be27b2254289f98a0514add547f0e2c93ec2ebb3236111832a702c4R1-R43))
*  Add `prepareToken` and `prepareCommonToken` functions to extract common logic of merging and preparing tag token ([link](https://github.com/ant-design/ant-design/pull/44512/files?diff=unified&w=0#diff-cb2262b77ae285e7da02504971f378bec871797a530abbf2a34f14fdb3d92f85L165-R157))
*  Update `genComponentStyleHook` call to use `prepareToken` and `prepareCommonToken` functions and to only generate base style ([link](https://github.com/ant-design/ant-design/pull/44512/files?diff=unified&w=0#diff-cb2262b77ae285e7da02504971f378bec871797a530abbf2a34f14fdb3d92f85L165-R157))
*  Update `genSubStyleComponent` function to accept same parameters as `genComponentStyleHook` and to set default options for sub-style components ([link](https://github.com/ant-design/ant-design/pull/44512/files?diff=unified&w=0#diff-e6843e3f6d078c2e0c399d9f79c5e0591073abc571cab4c1e71a9fac531baeacL187-R202))

### CODE REVIEW

**Review by kiner-tang — APPROVED** [review]

File: components/theme/util/genComponentStyleHook.ts [hunk 1]
```diff
@@ -184,13 +184,19 @@ export interface SubStyleComponentProps {
...
+export const genSubStyleComponent: <ComponentName extends OverrideComponent>(
   ...args: Parameters<typeof genComponentStyleHook<ComponentName>>
-): ComponentType<SubStyleComponentProps> {
-  const useStyle = genComponentStyleHook(...args);
+) => ComponentType<SubStyleComponentProps> = (componentName, styleFn, getDefaultToken, options) => {
+  const useStyle = genComponentStyleHook(componentName, styleFn, getDefaultToken, {
+    resetStyle: false,
+
+    // Sub Style should default after root one
+    order: -998,
```
Comment (afc163) [inline components/theme/util/genComponentStyleHook.ts:194]: order 的意义是啥？感觉研发样式的人会很难理解这个

**Comment by afc163** [comment]
关联一下：https://github.com/ant-design/ant-design/discussions/44309

File: components/theme/util/genComponentStyleHook.ts (hunk 1 above)
Comment (zombieJ) [reply components/theme/util/genComponentStyleHook.ts:194]: `@ant-design/cssinjs` 提供的插入顺序功能，antd 默认是 -999 以此防止时序上覆盖掉组件开发者的样式。例如 ProComponent 和 antd 在没有优先级的时候会出现 ProComponents 被后来添加的 antd 覆盖样式的情况。

Comment (zombieJ) [reply components/theme/util/genComponentStyleHook.ts:194]: > // Sub Style should default after root one

`genStyleComponent` 返回的组件都是通过子元素的形式进行动态插入，拆分样式都应该后置于基本样式。

**Comment by zxhycxq** [comment]
文案
 “现在只有当 **使用的使用** 才会生成它们” 
或许多了个 的使用 ？

**Pull request merged**